pyomo/contrib/mindtpy/util.py

```diff
@@ -606,3 +608,47 @@ def setup_solve_data(model, config):
                 direction=Suffix.IMPORT)
 
     return solve_data
+
+
+def copy_var_list_values_from_solution_pool(from_list, to_list, config, solver_model, var_map, solution_name,
+                                            skip_stale=False, skip_fixed=True,
+                                            ignore_integrality=False):
+    """Copy variable values from one list to another.
+
+    Rounds to Binary/Integer if neccessary
+    Sets to zero for NonNegativeReals if neccessary
+    """
+    for v_from, v_to in zip(from_list, to_list):
+        if skip_stale and v_from.stale:
+            continue  # Skip stale variable values.
+        if skip_fixed and v_to.is_fixed():
+            continue  # Skip fixed variables.
+        try:
+            if config.mip_solver == 'cplex_persistent':
+                var_val = solver_model.solution.pool.get_values(
+                    solution_name, var_map[v_from])
+            elif config.mip_solver == 'gurobi_persistent':
+                solver_model.setParam(
+                    gurobipy.GRB.Param.SolutionNumber, solution_name)
+                var_val = var_map[v_from].Xn
+            else:
+                config.logger(
+                    config.mip_solver + 'is not supported for solution pool in Mindtpy yet.')
+                raise NotImplementedError()
+            v_to.set_value(var_val)
+            if skip_stale:
+                v_to.stale = False
+        except ValueError as err:
+            err_msg = getattr(err, 'message', str(err))
+            rounded_val = int(round(var_val))
+            # Check to see if this is just a tolerance issue
+            if ignore_integrality \
+                    and v_to.is_integer():  # not v_to.is_continuous()
+                v_to.value = var_val
+            elif v_to.is_integer() and (fabs(var_val - rounded_val) <= config.integer_tolerance):  # not v_to.is_continuous()
```

Comment:
Change this to abs() since it is faster, type stable and works for this case better.